### PR TITLE
Try to fix the covaeralls issue

### PR DESCRIPTION
**Pull request by nunnatsa**
Since HCO moved to run sanity in github action, the coveralls report is wrong - relatives to
the current branch insted of to the master branch, so the coverage check does not show increasing
or decreasing.

This PR moves to use the shogo82148/actions-goveralls@v1 action in order to solve this issue.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
didn't work

File: .github/workflows/pr-sanity.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ name: Sanity Checks
 
 # Controls when the action will run.
 on:
+  push:
+    branches: [ main ]
```
Comment (erkanerol) [inline .github/workflows/pr-sanity.yaml:8]: @nunnatsa should not it be "master" ?

Comment (nunnatsa) [reply .github/workflows/pr-sanity.yaml]: fixed

Comment (orenc1) [reply .github/workflows/pr-sanity.yaml]: i would prefer to use native yaml format, but not critical for now.
```
on:
  push:
    branches:
    - master
```

File: hack/build-tests.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ if [ "${JOB_TYPE}" == "travis" ]; then
     go mod vendor
     PACKAGE_PATH="pkg/"
     mkdir -p coverprofiles
-    KUBEVIRT_CLIENT_GO_SCHEME_REGISTRATION_VERSION=v1 ginkgo -r -cover -outputdir=./coverprofiles -coverprofile=cover.coverprofile ${PACKAGE_PATH}
+    KUBEVIRT_CLIENT_GO_SCHEME_REGISTRATION_VERSION=v1 ginkgo -r -covermode atomic -outputdir=./coverprofiles -coverprofile=cover.coverprofile ${PACKAGE_PATH}
```
Comment (orenc1) [inline hack/build-tests.sh:17]: are you sure you don't need to still specify the `cover` flag to run the coverage tests?
```
  -cover
    	Run tests with coverage analysis, will generate coverage profiles with the package name in the current directory.
```

Comment (nunnatsa) [reply hack/build-tests.sh:17]: It's not needed if we are using the `-covermod` flag

Comment (orenc1) [reply hack/build-tests.sh:17]: /lgtm

**Pull request merged**